cli/whoami.ts

```diff
@@ -0,0 +1,43 @@
+import type {ArgumentsCamelCase} from 'yargs';
+import type {PublicApiUser} from '../helpers/external-api/v1';
+import {createCodaClient} from './helpers';
+import {formatEndpoint} from './helpers';
+import {getApiKey} from './config_storage';
+import {isResponseError} from '../helpers/external-api/coda';
+import {printAndExit} from '../testing/helpers';
+import {tryParseSystemError} from './errors';
+
+interface WhoamiArgs {
+  apiToken?: string;
+  codaApiEndpoint: string;
+}
+
+export async function handleWhoami({apiToken, codaApiEndpoint}: ArgumentsCamelCase<WhoamiArgs>) {
+  const formattedEndpoint = formatEndpoint(codaApiEndpoint);
+  if (!apiToken) {
+    apiToken = getApiKey(codaApiEndpoint);
+    if (!apiToken) {
+      return printAndExit('Missing API token. Please run `coda register` to register one.');
+    }
+  }
+
+  const client = createCodaClient(apiToken, formattedEndpoint);
+
+  try {
+    const response = await client.whoami();
+    return printAndExit(formatWhoami(response), 0);
+  } catch (err: any) {
+    if (isResponseError(err)) {
+      return printAndExit(`Invalid API token provided.`);
+    }
+
+    const errors = [`Unexpected error while checking owner of API token: ${err}`, tryParseSystemError(err)];
+    return printAndExit(errors.join('\n'));
+  }
+}
+
+export function formatWhoami(user: PublicApiUser) {
+  const {name, loginId, tokenName} = user;
+
+  return `You are ${name} (${loginId}) using token "${tokenName}"`;
```

Comment:
Ah right, we only expose whether it is scoped, and not what the scope is. I can't remember why we do this. What you have here is good, let's also just note if the token is scoped or not.

Reply:
Added a "scoped" or "non-scoped" adjective before the word "token" and updated the corresponding test